### DISCUSSION TITLE
zoom in and zoom out based upon nodes so that font size is inline with other part of web page

### DISCUSSION THREAD

**Discussion by dheerajmishradv** [xyflow/xyflow#4081] [Q&A]
I am creating reactflow dynamically based on the data I am getting from other services which is working fine. 
Example 1: If I have lots of node than the view looks good as font is ok as compared to other data in page.
Example 2: Now the problem is that if I have few nodes than the view is zoomed in and the font is tooo big as compared to font of other part of page and its look wierd.
so how can I make sure that the view does not get automatically zoomed to such extent that it looks very weird.
I have attached screenshot for reference
![reactFlow-example2](https://github.com/xyflow/xyflow/assets/144358323/60c4f2d4-609e-4417-8ac0-77a2cea442de)
![reactFlow-example1](https://github.com/xyflow/xyflow/assets/144358323/447911ff-8568-4e0c-a105-d0c854c1f3f8)

**Comment by moklick** [comment]
You can decrease [`maxZoom`](https://reactflow.dev/api-reference/react-flow#max-zoom) or pass `maxZoom` as a [`fitViewOptions`](https://reactflow.dev/api-reference/types/fit-view-options) attribute to address this.
